UPGRADE-6.4.md

```diff
@@ -1,12 +1,47 @@
 UPGRADE FROM 6.3 to 6.4
 =======================
 
+DependencyInjection
+-------------------
+
+ * Deprecate `ContainerAwareInterface` and `ContainerAwareTrait`, use dependency injection instead
+
+   Before:
+   ```php
+   class MyService implements ContainerAwareInterface
+   {
+       use ContainerAwareTrait;
+
+       // ...
+   }
+   ```
+
+   After:
+   ```php
+   class MyService
+   {
+       // Inject the container through the constructor...
+       public function __construct(private ContainerInterface $container)
+       {
+           $this->container = $container;
```

Comment:
cpp rulez

Reply:
Thinking of it again, isn't having CCP strange with the below example of the `#[Required]` attribute?

Reply:
why? Those are decorrelated

Reply:
It's true that it's decorrelated, it just made me feel like we were using 2 types of injection at the same time to write this example this way. But I'm aware it's a very subjective feeling :slightly_smiling_face: